staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -232,6 +233,7 @@ func NewCmdDrain(f cmdutil.Factory, ioStreams genericiooptions.IOStreams) *cobra
 	cmd.Flags().StringVarP(&o.drainer.PodSelector, "pod-selector", "", o.drainer.PodSelector, "Label selector to filter pods on the node")
 	cmd.Flags().BoolVar(&o.drainer.DisableEviction, "disable-eviction", o.drainer.DisableEviction, "Force drain to use delete, even if eviction is supported. This will bypass checking PodDisruptionBudgets, use with caution.")
 	cmd.Flags().IntVar(&o.drainer.SkipWaitForDeleteTimeoutSeconds, "skip-wait-for-delete-timeout", o.drainer.SkipWaitForDeleteTimeoutSeconds, "If pod DeletionTimestamp older than N seconds, skip waiting for the pod.  Seconds must be greater than 0 to skip.")
+	cmd.Flags().IntVar(&o.drainer.DelayBetweenNodeSeconds, "delay-between-nodes", o.drainer.DelayBetweenNodeSeconds, "When more than one node is targeted (e.g. via labels) this is the period of seconds to wait after each node completes draining.")
```

Comment:
Thanks for spending effort on this PR. However, as a sig-cli, we are very reluctant to add new flags, unless there is a strong justification and clear need. In that case, I don't think we'd want to introduce a new flag just for the purpose of adding a sleep.

Reply:
My pleasure!

The complexity for this change is quite low and the need is clearly defined in the documentation for the flag itself.

Here are some suggested justifications:

1. Roughly 25% of the flags for `drain` are related to controlling timeouts that give operators control of how a drain will be performed. The proposed flag is in good company, existing flags like `--timeout` are just as trivial to implement in a wrapper script and yet critical enough to daily life as an operator to be provided as a built-in affordance.
2. Clusters with absent or incomplete PDBs are extremely common. These clusters need to be upgraded the same as any other. Rapid successive draining of nodes can be overly aggressive, not leaving time for workloads to be scheduled and become ready before all are evicted. This option provides a simple method to alleviate this issue without wrapper scripts and one-off bookkeeping.
4. This feature is simple, provides good documentation for operators about pitfalls of targeting multiple nodes, and is fully tested.

As a current consumer of a substantially more complex wrapper script, I find the justifications for `delay-between--nodes` strong.

Reply:
The new flag does seem like it would be in good company with the other toggles related to draining.  Since the use case is for when nodes are label selected, that's something kubectl is positioned to know about. Whereas custom scripts would have to get nodes by selector and add sleeps themselves

Reply:
@ardaguclu Do you have any suggestions for how a conversation about the inclusion of this feature could continue? I would be glad to attend a sig-cli meeting to raise this more broadly if you feel there would be any value in doing so. If your response speaks for the entire sig-cli team, and that response is a final and unequivocal no, can you please close this PR?

Reply:
Sorry for the late response. It would be better to attend sig-cli bi-weekly meeting and bring this topic to get more feedback.